daemon/tests/harness/mocks/wallet.rs

```diff
@@ -1,93 +1,81 @@
-    psbt: Option<PartiallySignedTransaction>,
-    txid: Option<Txid>,
+/// Test Stub simulating the Wallet actor.
+/// Serves as an entrypoint for injected mock handlers.
+pub struct WalletActor {
+    pub mock: Arc<Mutex<dyn Wallet + Send>>,
 }
 
-impl Wallet {
-    pub fn with_party_params(mut self) -> Self {
-        self.party_params = true;
-        self
-    }
-
-    pub fn with_wallet_info(mut self) -> Self {
-        let s = Secp256k1::new();
-        let public_key = ecdsa::PublicKey::new(s.generate_keypair(&mut thread_rng()).1);
-        let address = bdk::bitcoin::Address::p2pkh(&public_key, bdk::bitcoin::Network::Testnet);
-
-        self.wallet_info = Some(WalletInfo {
-            balance: bdk::bitcoin::Amount::ONE_BTC,
-            address,
-            last_updated_at: Timestamp::now().unwrap(),
-        });
-
-        self
-    }
-
-    pub fn with_psbt(mut self, psbt: PartiallySignedTransaction) -> Self {
-        self.psbt = Some(psbt);
-        self
-    }
-
-    pub fn with_txid(mut self, txid: Txid) -> Self {
-        self.txid = Some(txid);
-        self
-    }
-}
-
-impl xtra::Actor for Wallet {}
+impl xtra::Actor for WalletActor {}
 
 #[xtra_productivity(message_impl = false)]
-impl Wallet {
+impl WalletActor {
     async fn handle(&mut self, msg: wallet::BuildPartyParams) -> Result<PartyParams> {
-        if self.party_params {
-            let mut rng = thread_rng();
-            let wallet = dummy_wallet(&mut rng, Amount::from_btc(0.4).unwrap(), 5).unwrap();
-
-            let party_params = wallet
-                .build_party_params(msg.amount, msg.identity_pk)
-                .unwrap();
-
-            return Ok(party_params);
-        }
-
-        panic!("Stub not configured to return party params")
+        self.mock.lock().await.build_party_params(msg)
     }
     async fn handle(&mut self, _msg: wallet::Sync) -> Result<WalletInfo> {
-        self.wallet_info
-            .clone()
-            .ok_or_else(|| anyhow!("Stub not configured to return WalletInfo"))
+        tracing::info!("We are handling the wallet sync msg");
+        self.mock.lock().await.sync()
     }
     async fn handle(&mut self, _msg: wallet::Sign) -> Result<PartiallySignedTransaction> {
-        self.psbt
-            .clone()
-            .ok_or_else(|| anyhow!("Stub not configured to return PartiallySignedTransaction"))
+        self.mock.lock().await.sign()
     }
     async fn handle(&mut self, _msg: wallet::TryBroadcastTransaction) -> Result<Txid> {
-        self.txid
-            .ok_or_else(|| anyhow!("Stub not configured to return Txid"))
+        self.mock.lock().await.broadcast()
     }
 }
 
-impl Default for Wallet {
-    fn default() -> Self {
-        Wallet {
-            party_params: false,
-            wallet_info: None,
-            psbt: None,
-            txid: None,
-        }
+#[automock]
+pub trait Wallet {
+    fn build_party_params(&mut self, _msg: wallet::BuildPartyParams) -> Result<PartyParams> {
+        todo!()
```

Comment:
```suggestion
        unreachable!("Generated code that will be filled in the test")
```


I find `todo!` misleading here - does `mockall` enforce that? I think`unreachable!` would be easier to understand. 
Would be nice if we can add a message so it becomes more clear what this is.

Reply:
good idea, thanks!